frontend/src/lib/lemon-ui/LemonTree/LemonTree.tsx

```diff
@@ -308,44 +371,120 @@ const LemonTreeNode = forwardRef<HTMLDivElement, LemonTreeNodeProps>(
+                                                                    placement="top-start"
+                                                                    className="w-fit"
                                                                 >
-                                                                    {displayName}
-                                                                </span>
-                                                            )}
-
-                                                            {/* Loading state */}
-                                                            {item.record?.loading && <Spinner className="ml-1" />}
-
-                                                            {/* Unapplied state */}
-                                                            {item.record?.unapplied && (
-                                                                <IconUpload className="ml-1 text-warning" />
-                                                            )}
-                                                        </>
-                                                    ) : (
-                                                        <span className={cn('truncate', isFolder && 'font-semibold')}>
-                                                            {displayName}
-                                                        </span>
-                                                    )}
+                                                                    <span>{displayName}</span>
+                                                                </Tooltip>
+                                                            </span>
+                                                        )}
+
+                                                        {/* Loading state */}
+                                                        {item.record?.loading && <Spinner className="ml-1" />}
+
+                                                        {/* Unapplied state */}
+                                                        {item.record?.unapplied && (
+                                                            <IconUpload className="ml-1 text-warning" />
+                                                        )}
+                                                        {mode === 'table' &&
+                                                            tableViewKeys?.headers.slice(1).map((header, index) => {
+                                                                return (
+                                                                    <span
+                                                                        key={header.key}
+                                                                        className="truncate text-left"
+                                                                    >
+                                                                        {(() => {
```

Comment:
You can avoid this additional nested immediately executing function as you're already in a ".map" above.